### DISCUSSION TITLE
Using a route-generation library that plays better with ESM imports

### DISCUSSION THREAD

**Discussion by ElMassimo** [ledermann/pingcrm#515] [Ideas]
Hi @ledermann, first of all, thanks for writing and sharing this demo, it works great!

I wanted to ask you if you would be interested in accepting [this pull request](https://github.com/ElMassimo/pingcrm-vite/pull/2).

As I mention in its description, it allows to go from:

```js
this.$inertia.put(this.$routes.restore_contact(this.contact.id))
```

to:

```js
ContactsRequests.restore(this.contact)
```

and requires no additional runtime dependencies (plus the generated code is smaller and code-splittable).

Let me know what you think, didn't want to open a pull request without first asking 😃

**Comment by ledermann** [comment]
Thanks, @ElMassimo, I'm glad this little project is helpful - and it's great to see that you build a Vite fork :) 

However, I'm not quite convinced about [js_from_routes](https://github.com/ElMassimo/js_from_routes) yet. Your PR contain's LOTS of code, which of course seems mainly be auto generated. But I'm afraid of all the setup code required in the `.vue` component files.

Currently I still like the old [js-routes](https://github.com/railsware/js-routes) way and see no reason why to switch. Better ESM imports is a thing, but IMHO it' not a big deal to load all routes in one chunk. The `routes.js` for PingCRM is about 10KB (minified), another project of mine has 13KB routes. Splitting is IMHO not required. 

Nevertheless, thanks a lot for your interest in this project - and please tell me If I totally missed something about `js_from_routes`.

**Reply by ElMassimo** [reply]
No problem, just mentioning the approach. Have in mind that the "LOTS" of code is actually less lines of code than the `routes.js` 😉 

About route splitting, that's a fair point, in smaller projects it might be fine to have all routes in the same file.
